frontend/simple/js/state.js

```diff
@@ -167,38 +170,36 @@ const actions = {
       //       and throw an exception and write a log message.
       return console.error(`NOT EXPECTING EVENT!`, contractId, entry)
     }
-
     const type = entry.type
-    entry = Events[type].fromObject(entry, hash)
-    if (entry instanceof Events.HashableContract) {
+    entry = Events[type] ? Events[type].fromObject(entry, hash) : contracts[type].fromObject(entry, hash)
+    if (Events.HashableContract.prototype.isPrototypeOf(entry)) {
```

Comment:
It was not detecting properly in two places after the split with the instanceof  so I rewrote it with prototypeof